tpm/attestation/client.go

```diff
@@ -0,0 +1,282 @@
+package attestation
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"time"
+
+	"go.step.sm/crypto/tpm"
+
+	"github.com/smallstep/go-attestation/attest"
+)
+
+type Client struct {
+	client  *http.Client
+	baseURL *url.URL
+}
+
+type Options struct {
+	rootCAs  *x509.CertPool
+	insecure bool
+}
+
+type Option func(o *Options) error
+
+// WithRootsFile can be used to set the trusted roots when
+// setting up a TLS connection. An empty filename will
+// be ignored.
+func WithRootsFile(filename string) Option {
+	return func(o *Options) error {
+		if filename == "" {
+			return nil
+		}
+		data, err := os.ReadFile(filename)
+		if err != nil {
+			return fmt.Errorf("failed reading %q: %w", filename, err)
+		}
+		pool := x509.NewCertPool()
+		if !pool.AppendCertsFromPEM(data) {
+			return fmt.Errorf("failed parsing %q: no certificates found", filename)
+		}
+		o.rootCAs = pool
+		return nil
+	}
+}
+
+// WithInsecure disables TLS server certificate chain checking.
+// In general this shouldn't be used, but it can be of use in
+// during development and testing.
+func WithInsecure() Option {
+	return func(o *Options) error {
+		o.insecure = true
+		return nil
+	}
+}
+
+// NewClient creates a new Client that can be used to perform remote
+// attestation.
+func NewClient(tpmAttestationCABaseURL string, options ...Option) (*Client, error) {
+	u, err := url.Parse(tpmAttestationCABaseURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed parsing attestation CA base URL: %w", err)
+	}
+
+	opts := &Options{}
+	for _, o := range options {
+		if err := o(opts); err != nil {
+			return nil, fmt.Errorf("failed applying option to attestation client: %w", err)
+		}
+	}
+
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			TLSClientConfig: &tls.Config{
+				RootCAs:            opts.rootCAs,
+				InsecureSkipVerify: opts.insecure, //nolint:gosec // intentional insecure if provided as option
+			},
+		},
```

Comment:
We can clone the DefaultTransport that has good default and add the `TLSClientConfig`:

```go
transport := http.DefaultTransport.(*http.Transport).Clone()
transport.TLSClientConfig = &tls.Config{
    // ..
}
```